packages/hawtio/src/plugins/camel/CamelTreeView.tsx

```diff
@@ -29,17 +30,28 @@ enum ExpansionValue {
 export const CamelTreeView: React.FunctionComponent = () => {
   const { tree, selectedNode, setSelectedNode } = useContext(CamelContext)
   const [expanded, setExpanded] = useState(ExpansionValue.Default)
-  const [filteredTree, setFilteredTree] = useState(tree.getTree())
   const navigate = useNavigate()
 
+  const deriveTree = useCallback((): MBeanNode[] => {
+    const t = tree.getTree()
+    if (t.length === 0) return t
+
+    if (!ccs.isDomainNode(t[0])) return t
+
+    // Exclude the domain node from the tree display
+    return t[0].getChildren()
+  }, [tree])
```

Comment:
Why do we need this derivation here instead of doing so inside the `useContext`?  Is there any reason we need to bring the skipped root node into the react component?